packages/ui/src/accounts/components/AccountItem/components/lockItems/ProposalLockItem.tsx

```diff
@@ -11,12 +12,24 @@ import { LockLinkButton } from '../LockLinkButton'
 import { LockDetailsProps } from '../types'
 
 export const ProposalLockItem = ({ lock, address, isRecoverable }: LockDetailsProps) => {
+  const { api } = useApi()
   const { data } = useGetLatestProposalByMemberIdQuery({ variables: { lockAccount: address } })
   const proposal = data?.proposals[0]
   const eventData = proposal?.createdInEvent
   const createdInEvent = eventData && asBlock(eventData)
 
-  const recoveryTime = faker.date.soon(1).toISOString()
+  const recoveryTime = useMemo(() => {
+    if (!eventData || !api || isRecoverable) {
+      return null
+    }
+    const startTime = new Date(eventData.createdAt).getTime()
```

Comment:
This logic works for almost all cases, but sometime proposal have a trigger execution block, in which case there should be one more if:
```ts
if (proposal.exactExecutionBlock) {
  const startTime = Date.now()
  const duration = proposal.exactExecutionBlock - currentBlockNumber
  return new Date(startTime + duration * MILLISECONDS_PER_BLOCK).toISOString()
}